impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultInterpolator.java

```diff
@@ -283,7 +283,7 @@ private static String doSubstVars(
     }
 
     private static String processSubstitution(
-            String variable,
+            final String variable,
```

Comment:
boilerplate tradeoff naturally evolves to https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#avoidreassigningparameters.

Reply:
assuming you mean the explicit `final` declaration. Params should be **treated** `final by default` to avoid coupling thus bugs. 

- https://sonarsource.github.io/rspec/#/rspec/S1226